client/src/components/columns.jsx

```diff
@@ -1,55 +1,38 @@
 import React, { Component } from "react";
 import axios from "axios";
-let arrhappy = [];
-let arrsad = [];
-const api = "http://localhost:3001/sites";
 
 class Columns extends Component {
   state = {
-    data: [{ id: 0, name: "", urls: ["", ""] }],
     happy: [],
     sad: [],
     faking: []
   };
   componentDidMount() {
     this.getDataFromDb();
-    this.parseData(this.state.data);
   }
 
   getDataFromDb = () => {
-    axios.get(api).then(result => {
-      console.log(result.data);
-      let variable = [];
-      variable = result.data;
-      this.setState({ data: variable });
+    axios.get("/sites/check").then(result => {
+      this.updateStatus(result.data);
     });
   };
 
-  parseData = data => {
-    data.map(site => {
-      this.checkStatus(site);
-    });
-  };
 
-  checkStatus(site) {
-    site.urls.map(url =>
-      axios.get(url).then(response => {
-        console.log(response);
-        if (response.status !== 200) {
-          arrsad.push(site.name);
-          console.log(site);
-        } else {
-          arrhappy.push(site.name);
-          console.log(site);
-        }
-      })
-    );
-    this.update();
+  updateStatus(status) {
+    const result = Object.keys(status).map(key => {
+      const statuses = new Set(Object.values(status[key]));
+      // is happy if it's always 200, sad if there is no 200, faking if it's mix
+      const state = statuses.has(200)?(statuses.size===1?"happy":"faking"):"sad";
+      return [key, state]
+    }).reduce((acc, [name, state]) => {
+      // group names in happy, sad, faking categories
+      acc[state].push(name);
+      return acc;
+    }, {happy: [], sad: [], faking: []});
+    this.setState(result);
   }
 
-  update() {
-    this.setState({ happy: arrhappy });
```

Comment:
this will do nothing - `arrhappy` was defined in the global scope - so react will ignore this state update  - from react perspective it does not change